From dd5a82b35570e454ada3b3c5f42742865e02bc93 Mon Sep 17 00:00:00 2001
From: Jeremy Rocher <jeremy.rocher@intel.com>
Date: Mon, 17 Jun 2013 19:37:56 +0200
Subject: [PORT FROM MAIN] Add Valgrind in eng builds

BZ: 60845

Create phony package for Valgrind and add valgring_pack in eng builds.

Change-Id: Ia546ae903f3f6fb2c237cef4ebde35555f1b602a
Orig-Change-Id: I2b18b89ca61b2fc8239d9f8ab1231aad291d44f5
Signed-off-by: Jeremy Rocher <jeremy.rocher@intel.com>
---
 common.mk        |    5 +++++
 debug/Android.mk |   26 ++++++++++++++++++++++++++
 2 files changed, 31 insertions(+), 0 deletions(-)
 create mode 100644 debug/Android.mk

diff --git a/common.mk b/common.mk
index 646b3b2..257061d 100644
--- a/common.mk
+++ b/common.mk
@@ -93,6 +93,11 @@ PRODUCT_COPY_FILES += \
         $(PERMISSIONS_PATH)/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
         $(PERMISSIONS_PATH)/android.hardware.camera.autofocus.xml:system/etc/permissions/android.hardware.camera.autofocus.xml
 
+# Valgrind for eng build variant
+ifneq ($(filter $(TARGET_BUILD_VARIANT),eng),)
+PRODUCT_PACKAGES += \
+    valgrind_pack
+endif
 
 # Power Debug Tools
 -include vendor/intel/hardware/PRIVATE/platform_test/power-debug/power-debug.mk
diff --git a/debug/Android.mk b/debug/Android.mk
new file mode 100644
index 0000000..bd8150b
--- /dev/null
+++ b/debug/Android.mk
@@ -0,0 +1,26 @@
+LOCAL_PATH := $(call my-dir)
+
+# Valgrind
+include $(CLEAR_VARS)
+LOCAL_MODULE := valgrind_pack
+LOCAL_MODULE_TAGS := optional
+LOCAL_REQUIRED_MODULES := \
+    libvex-$(TARGET_ARCH)-linux \
+    libcoregrind-$(TARGET_ARCH)-linux \
+    libreplacemalloc_toolpreload-$(TARGET_ARCH)-linux \
+    vgpreload_core-$(TARGET_ARCH)-linux \
+    memcheck-$(TARGET_ARCH)-linux \
+    vgpreload_memcheck-$(TARGET_ARCH)-linux \
+    cachegrind-$(TARGET_ARCH)-linux \
+    callgrind-$(TARGET_ARCH)-linux \
+    helgrind-$(TARGET_ARCH)-linux \
+    vgpreload_helgrind-$(TARGET_ARCH)-linux \
+    drd-$(TARGET_ARCH)-linux \
+    vgpreload_drd-$(TARGET_ARCH)-linux \
+    massif-$(TARGET_ARCH)-linux \
+    vgpreload_massif-$(TARGET_ARCH)-linux \
+    none-$(TARGET_ARCH)-linux \
+    valgrind \
+    default.supp
+include $(BUILD_PHONY_PACKAGE)
+
-- 
1.7.4.1

